examples/react-native/__tests__/ts/realm-database/change-an-object-model-test.tsx

```diff
@@ -1,23 +1,80 @@
+import React from 'react';
+import {View, Text} from 'react-native';
 import Realm from 'realm';
 import {createRealmContext} from '@realm/react';
-import Person from '../Models/Person';
+import {render} from '@testing-library/react-native';
+
+// Replace incremented schema versions in Bluehawk output. The incrementing
+// is necessary for testing, but confusing outside that context. Also remove
+// useRealm instantiation, for the same reason.
+// :replace-start: {
+//    "terms": {
+//       "schemaVersion: 3": "schemaVersion: 2",
+//       "schemaVersion: 4": "schemaVersion: 2",
+//       "schemaVersion: 5": "schemaVersion: 2",
+//       "oldRealm.schemaVersion < 4": "oldRealm.schemaVersion < 2",
+//       "oldRealm.schemaVersion < 5": "oldRealm.schemaVersion < 2",
+//        "RealmProvider, useRealm": "RealmProvider"
+//    }
+// }
+
+class Person extends Realm.Object<Person> {
```

Comment:
```suggestion
class Person extends Realm.Object {
```

I'm going to clear this up with the team now, but I believe since we are not documenting creating objects with the constructor, we might be able to remove this generic declaration

Reply:
I looked into the types found some small typing issues with not including `<Person>`, though it seems possible that we could fix our types to make it work without.
One thing happens is that when you do:
```
person.addListener((obj, changes) => {
 //...
});
```
That `obj` would be typed as `unknown` instead of `Person`.  `linkingObject` method is also effected by this.

Another thing to keep in mind is that we have an `RequiredTypes` generic as well (which answers one of your questions elsewhere).
```
class Person extends Realm.Object<Person, 'name' | 'age'> {
//...
```
This is used since currently when creating an object, all the fields are optional.  The `RequiredTypes` say which fields are non-optional and will show a type error if they are not included.  For the example above, you would get a type error if name or age were not defined when trying to create an object.  Unfortunately, it's only doing this on the "constructor" creation (using `new Person(...)`) and no on the `create` method (using `realm.create(Person...)`).  

That being said, I suggest, we keep the generic in.  I'll make open an issue to clean this up.

Reply:
Thanks for sharing this info! We'll keep the generic then. I think I'll also experiment with a dedicated TypeScript page that explains things like this.